src/visGeometry/index.ts

```diff
@@ -1086,18 +1084,44 @@ class VisGeometry {
         }
     }
 
-    public addNewColor(color: number | string): void {
+    private indexOfColor(color: number[]) {
+        const colorArray = this.colorsData;
+        const colorToCheck = map(color, (num) => round(num, 6));
+        for (let i = 0; i < colorArray.length - 3; i += 4) {
+            const index = i / 4;
+            const currentColor = [
+                round(this.colorsData[i], 6),
+                round(this.colorsData[i + 1], 6),
+                round(this.colorsData[i + 2], 6),
+                this.colorsData[i + 3],
+            ];
+            if (isEqual(currentColor, colorToCheck)) {
+                return index;
+            }
+        }
+        return -1;
+    }
+
+    public addNewColor(color: number | string): number {
         const colorNumber = convertColorStringToNumber(color);
         const newColor = [
             ((colorNumber & 0x00ff0000) >> 16) / 255.0,
             ((colorNumber & 0x0000ff00) >> 8) / 255.0,
             ((colorNumber & 0x000000ff) >> 0) / 255.0,
             1.0,
         ];
+        const currentIndex = this.indexOfColor(newColor);
```

Comment:
when does addNewColor get called?  Do you always know the agent type when you are calling it?

Reply:
it's just in the case that the user changed the color of an agent to a color already in the data array

Reply:
These get called in changeAgentsColor in viewport, which takes agent, tags, and color in the form of colorChanges.  The function call that needs the IDs returned from addNewColor is setColorForIds in visGeo.